backend/uclapi/workspaces/occupeye.py

```diff
@@ -0,0 +1,348 @@
+            self.r.llen("occupeye:surveys") - 1
+        )
+
+        surveys = []
+
+        for id in survey_ids:
+            survey_data = self.r.hgetall("occupeye:surveys:" + id)
+            survey = {
+                "id": int(survey_data["id"]),
+                "name": survey_data["name"],
+                "active": self._str2bool(survey_data["active"]),
+                "start_time": survey_data["start_time"],
+                "end_time": survey_data["end_time"]
+            }
+            survey_map_ids_list = "occupeye:surveys:{}:maps".format(
+                id
+            )
+            survey_map_ids = self.r.lrange(
+                survey_map_ids_list,
+                0,
+                self.r.llen(survey_map_ids_list)
+            )
+            survey_maps = []
+            for survey_map_id in survey_map_ids:
+                survey_map = self.r.hgetall(
+                    "occupeye:surveys:{}:maps:{}".format(
+                        id,
+                        survey_map_id
+                    )
+                )
+                survey_maps.append(
+                    {
+                       "id": int(survey_map["id"]),
+                       "name": survey_map["name"],
+                       "image_id": int(survey_map["image_id"])
+                    }
+                )
+            survey["maps"] = survey_maps
+            surveys.append(survey)
+        return surveys
+
+    def _cache_image(self, image_id):
+        """
+        Downloads map images from the API and stores their
+        base64 representation and associated data type in Redis.
+        """
+        headers = {
+            "Authorization": self.get_bearer_token()
+        }
+        url = "{}/api/images/{}?deployment={}".format(
+            self.base_url,
+            image_id,
+            self.deployment_name
+        )
+        try:
+            request = requests.get(
+                url=url,
+                headers=headers,
+                stream=True
+            )
+            content_type = request.headers['Content-Type']
+        except:
+            raise BadOccupEyeRequest
+
+        raw_image = request.content
+        image_b64 = b64encode(raw_image)
+
+        pipeline = self.r.pipeline()
+        pipeline.set(
+            "occupeye:image:{}:base64".format(image_id),
+            image_b64
+        )
+        pipeline.expire(
+            "occupeye:image:{}:base64".format(image_id),
+            self.IMAGE_TTL
+        )
+        pipeline.set(
+            "occupeye:image:{}:content_type".format(image_id),
+            content_type
+        )
+        pipeline.expire(
+            "occupeye:image:{}:content_type".format(image_id),
+            self.IMAGE_TTL
+        )
+        pipeline.execute()
+
+    def get_image(self, image_id):
+        """
+        Pulls an Image ID requested by the user from Redis.
+        """
+        try:
+            image_id_int = int(image_id)
+            image_id = str(image_id_int)

```

Comment:
what is the purpose of this?

Reply:
It's to make sure people don't pass in bad image id values, otherwise they could inject into our Redis database using user input. That would be baaad

Reply:
Can't you just do this:

```
if not image_id.isdigit():
    raise BadOccupEyeRequest()

if not self.r.exists(...):
    ...
```